transport/server.cc

```diff
@@ -207,11 +207,68 @@ cql_server::cql_server(distributed<cql3::query_processor>& qp, auth::service& au
         sm::make_derive("requests_shed", _stats.requests_shed,
                         sm::description("Holds an incrementing counter with the requests that were shed due to overload (threshold configured via max_concurrent_requests_per_shard). "
                                             "The first derivative of this value shows how often we shed requests due to overload in the \"CQL transport\" component.")),
-       sm::make_gauge("requests_memory_available", [this] { return _memory_available.current(); },
+        sm::make_gauge("requests_memory_available", [this] { return _memory_available.current(); },
                         sm::description(
                             seastar::format("Holds the amount of available memory for admitting new requests (max is {}B)."
                                             "Zero value indicates that our bottleneck is memory and more specifically - the memory quota allocated for the \"CQL transport\" component.", _max_request_size))),
+        
+        sm::make_derive("server_errors", _stats.errors[exceptions::exception_code::SERVER_ERROR],
+                        sm::description("Counts the total number of raised CQL SERVER errors.")),
+
+        sm::make_derive("protocol_errors", _stats.errors[exceptions::exception_code::PROTOCOL_ERROR],
+                        sm::description("Counts the total number of raised CQL PROTOCOL errors.")),
+
+        sm::make_derive("authentication_errors", _stats.errors[exceptions::exception_code::BAD_CREDENTIALS],
+                        sm::description("Counts the total number of raised CQL AUTHENTICATION errors.")),
```

Comment:
Nitpick: this is the only case in this patch where the message differs from the name of the error code. I think here too you should write "BAD_CREDENTIALS" in te description, not "AUTHENTICATION". Unless the name BAD_CREDENTIAL itself is a mistake?

Reply:
I used the name of the error code `0x0100` specified in the CQL v4 Protocol (https://github.com/apache/cassandra/blob/trunk/doc/native_protocol_v4.spec - section 9). Don't know whether there's a reason our name for it differs.
Maybe somebody knows some more context?